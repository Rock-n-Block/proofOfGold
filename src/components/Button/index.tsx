import React from 'react';
import { Button as BtnAntd } from 'antd';
import classNames from 'classnames';

interface ButtonProps {
  size?: 'lg' | 'sm';
  centered?: boolean;
  onClick?: () => void;
}

const Button: React.FC<React.HTMLAttributes<HTMLDivElement> & ButtonProps> = ({
  children,
  className,
  size,
  centered,
  onClick,
}) => {
  return (
    <BtnAntd
      onClick={onClick}
      className={classNames(className, 'text-md btn', `btn-${size}`, {
        centered: centered,
      })}>
      {children}
    </BtnAntd>
  );
};

export default Button;
