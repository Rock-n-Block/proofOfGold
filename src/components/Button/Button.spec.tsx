import React from 'react';
import { shallow } from 'enzyme';
import Button from './index';

const setUp = (props?: any) => shallow(<Button {...props}></Button>);

describe('Button Component', () => {
  it('Renders Button', () => {
    const button = setUp();
    expect(button).toMatchSnapshot();
  });

  it('Render Button with size class', () => {
    const button = setUp({
      size: 'lg',
    });
    expect(button.hasClass('btn-lg')).toBe(true);
  });
  it('Render Button with colorScheme class', () => {
    const button = setUp({
      colorScheme: 'black',
    });
    expect(button.hasClass('btn-black')).toBe(true);
  });
  it('Render Button with centered class', () => {
    const button = setUp({
      centered: true,
    });
    expect(button.hasClass('centered')).toBe(true);
  });
  it('Render disabled Button', () => {
    const button = setUp({
      disabled: true,
    });
    expect(button.props().disabled).toBe(true);
  });
  it('Render click Button', () => {
    const result: any =
      Button.defaultProps &&
      Button.defaultProps.onClick &&
      Button.defaultProps.onClick();

    expect(result).toBe(undefined);
  });
});
