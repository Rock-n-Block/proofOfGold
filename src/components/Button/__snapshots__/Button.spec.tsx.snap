// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Renders Button 1`] = `
<Button
  className="text-md btn btn-md btn-gold"
  disabled={false}
/>
`;
